app/controllers/user.js

```diff
@@ -0,0 +1,37 @@
+'use strict';
+
+const bcrypt = require('bcrypt'),
+  saltRounds = 10,
+  salt = bcrypt.genSaltSync(saltRounds),
+  { validateEmail, validatePassword } = require('./validations'),
+  errors = require('../errors'),
+  User = require('../models').users;
+
+exports.singUp = (req, res, next) => {
+  const signErrors = [];
+  const user = req.body
+    ? {
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        email: req.body.email,
+        password: req.body.password
+      }
+    : {};
+
+  if (!validateEmail(user.email)) signErrors.push(errors.emailError());
+
+  if (!validatePassword(user.password)) signErrors.push(errors.passwordError());
+
+  if (signErrors.length) throw errors.signupError(signErrors);
```

Comment:
Use a `try/catch` and send by `next`, for more clarity (you can use `await` for existing promises)